### PR TITLE
[DEV-2242] distance: add haversine distance calculation

### PR DESCRIPTION
**Pull request by jevonyeoh**
## Description
Add support to calculate `haversine` distances between 2 lat/lon points.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/core/distance.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        Series representing the longitude of the second point
+
+    Returns
+    -------
+    Series
+
+    Raises
+    ------
+    ValueError
+        if any of the Series are not of FLOAT type
```
Comment (YS-L) [inline featurebyte/core/distance.py:36]: nit: should be "numeric" here since integers are also allowed?

File: featurebyte/core/distance.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            f"lon_series_2: {lon_series_2.dtype},\n"
+        )
+
+    node = lat_series_1.graph.add_operation(
+        node_type=NodeType.HAVERSINE,
+        node_params={},
+        node_output_type=NodeOutputType.SERIES,
+        input_nodes=[lat_series_1.node, lon_series_1.node, lat_series_2.node, lon_series_2.node],
+    )
+    return type(lat_series_1)(
```
Comment (YS-L) [inline featurebyte/core/distance.py:65]: Would be good to validate that all the inputs are either all columns or all features but not a mix.

File: featurebyte/query_graph/sql/adapter/base.py [hunk 3]
```diff
@@ -693,3 +693,109 @@ def get_percentile_expr(cls, input_expr: Expression, quantile: float) -> Express
...
+    def _radian_expr(expr: Expression) -> Expression:
+        return expressions.Anonymous(this="RADIANS", expressions=[expr])
+
+    @staticmethod
+    def _square_expr(expr: Expression) -> Expression:
+        return expressions.Pow(this=expr, power=make_literal_value(2))
+
+    @staticmethod
+    def _sqrt_expr(expr: Expression) -> Expression:
+        return expressions.Anonymous(this="SQRT", expressions=[expr])
```
Comment (YS-L) [inline featurebyte/query_graph/sql/adapter/base.py:707]: sqlglot has `expressions.Sqrt`, would be good to use it.

**Review by YS-L — APPROVED** [review]

File: featurebyte/core/distance.py (hunk 2 above)
Comment (jevonyeoh) [reply featurebyte/core/distance.py:65]: will add

**Pull request merged**